test/check-machines-create

```diff
@@ -908,9 +923,16 @@ class TestMachinesCreate(VirtualMachinesCase):
             # For the mocked fedora URL installation wait to see that the OS section is automatically filled
             if self.sourceType == "url" and self.location == TestMachinesCreate.TestCreateConfig.TREE_URL:
                 b.wait_attr("#os-select-group input", "value", TestMachinesCreate.TestCreateConfig.FEDORA_28)
-            elif self.os_name:
-                b.click("#os-select-group > div button")
+
+            if self.os_name:
+                # After OS input already has a value, the expand button need selector with more details
+                b.click("#os-select-group > div button.pf-c-select__toggle-button")
                 b.click("#os-select li:contains('{0}') button".format(self.os_name))
+                b.wait_attr("#os-select-group input", "value", self.os_name)
+                if self.expected_os_name and self.sourceType == 'url':
+                    b.set_input_text("#source-url", self.location)
```

Comment:
Sorry for the iteration again.
This set_input_text("#source-url", self.location) is not needed.

It's already done in https://github.com/cockpit-project/cockpit-machines/pull/398/commits/d608cceef38a94702ac55fec87f4bec1fc81ec3d#diff-b284ac42752f40597c4dd12d74e16f6ed82ad5e12d68f0437f242f5ae20bbaa5R916 a few lines above.

Same for the condition, you can drop the self.sourceType == 'url'.

The checks for expected_os_name is enough.